qiskit/transpiler/passes/routing/commuting_2q_gate_routing/commuting_2q_gate_router.py

```diff
@@ -339,7 +339,7 @@ def swap_decompose(
             # Apply SWAP gates
             if i < max_distance:
                 for swap in swap_strategy.swap_layer(i):
-                    (j, k) = [trivial_layout.get_physical_bits()[vertex] for vertex in swap]
+                    (j, k) = (trivial_layout.get_physical_bits()[vertex] for vertex in swap)
```

Comment:
It's pretty irrelevant, but swapping small list comprehensions for generator expressions when building tuples is usually actually worse for runtime and memory.  List comprehensions produce objects of known width, and CPython is super optimised around creating them.  Generator expressions have to use more general paths, and in more recent Pythons, list comprehensions can even use one fewer stack frame per entry than them.

Reply:
It's maybe worth benchmarking these things now and then. OTOH it takes time and letting slide a small performance hit resulting from an erroneous attempt to get a small performance gain is ok; or maybe it can be disabled in a config file. 
 
Yesterday, I benchmarked allocating a new `list` in a loop rather than calling `.clear()`. The allocating code was faster over all variable ranges I tested.

Reply:
Another point. In Julia which one is faster varies from version to version. I wouldn't be surprised if Python is similar.

Reply:
Reverted